MantidQt/MantidWidgets/src/InstrumentView/InstrumentActor.cpp

```diff
@@ -757,9 +761,9 @@ void InstrumentActor::setupPickColors() {
 void InstrumentActor::cacheDetPos() const {
   if (m_detPos.size() != m_detIDs.size()) {
     m_detPos.clear();
-    for (size_t i = 0; i < m_detIDs.size(); i++) {
-      IDetector_const_sptr det = this->getDetector(i);
-      m_detPos.push_back(det->getPos());
+    for (size_t pickID = 0; pickID < m_detIDs.size(); pickID++) {
```

Comment:
`m_detIDs` contains all detector ids. I can try to replace them with the indices.
`pickID`s are like detector indices but I am not sure at the moment if they can be easily replaced with them without significant restructuring of the code. It's one of the things I wanted to think about.